hpc0.qmd

```diff
@@ -103,195 +115,513 @@ ssh <username>@arc4.leeds.ac.uk
 
-### Solution
+### Solutions
 
-1:
+1. `i++` is shorthand for 'i=i+1'
 ```{.bash code-line-numbers=false}
-[user@host ~]$ for (( i=1 ; i<=5 ; i++ ))
+[user@host ~]$ for ((i=1;i<6;i++))
 > do
 > mkdir calculation_$i
 > done
 [user@host ~]$
 ```
-
-2:
+\
+2. Two commands are executed in each loop. A new directory is created and then the working directory is changed to the new directory. On the next iteration, the new directory is then created inside the previous one.
 ```{.bash code-line-numbers=false}
-[user@host ~]$ for (( i=1 ; i<=5 ; i++ ))
-> do
+[user@host ~]$ for i in {1..5}; do
 > mkdir calculation_$i
 > cd calculation_$i
 > done
 [user@host calculation_5]$
 ```
 
-In the second case, two commands are executed each time you go through the loop. A new directory is created and
-then the working directory is changed to the new directory.
-
-On the next iteration, the new directory is then created inside the previous one.
-
 :::
 
-## Exercise 6 {.smaller}
+## Exercise 8 {.smaller}
 
 ::: {.panel-tabset}
 
 ### Problem
 
-1. In the wildcards directory, create a variable called files listing all of the text files. 
-2. Loop through this list and print out the first line from each file.
+Print out the first line of each file in the `wildcards` directory.
 
 ### Solution
 
-1:
 ```{.bash code-line-numbers=false}
-[user@host wildcards]$ files=$( ls *.txt )
-```
-
-2:
-
-```{.bash code-line-numbers=false}
-[user@host wildcards]$ for f in $files
+[user@host wildcards]$ for f in *.txt
 > do
-> head -1 $f
+> head -n 1 $f
 > done
 ```
 
 :::
 
-## Exercise 7 {.smaller}
+
+## Exercise 9 {.smaller}
 
 ::: {.panel-tabset}
 
 ### Problem
 
-1. What will this command print to the screen?
+What will this command print to the screen?
 
 ```{.bash code-line-numbers=false}
-[user@host wildcards]$ for filename in *.txt
+[user@host wildcards]$ for f in *.txt
 > do
-> echo $filename
-> cat $filename > new-file.txt
+> echo $f
+> cat $f > new-file.txt
 > done
 ```
 
 \
-2. What will the contents of `new-file.txt` be and why?
-
+What will the contents of `new-file.txt` be and why?
 
 ### Solution
 
-1: 
-
-- The name of each `.txt` file in the current directory
-- On each iteration the echo command prints a different file name
-
-2:
+- The command will print the name of each `.txt` file in the current directory
+- On each iteration the `echo` command prints a different file name
 
 - The content of `new-file.txt` will be the same as `xyz.txt`
 - On each iteration the contents of a different file are written to `new-file.txt`, overwriting whatever was written on the previous iteration
-- This is because we used `>` to write the output of cat to `new-file.txt`
+- This is because `>` overwrites the output of cat to `new-file.txt` for each element in the sequence.
 - Using `>>` instead would result in each file's content being appended to the end of `new-file.txt`
 
 :::
 
+
+## Shell Scripts
```

Comment:
Slide feels visually cluttered.  Can we reword/rejig a bit?

Reply:
Yes